src/ui/settingsPage/settingsPage.js

```diff
@@ -0,0 +1,186 @@
+  }
+  filterInput = createRef();
+
+  render() {
+    // Step 1 Filter the List, so only items that include the website are here
+    const filteredList = filter(this.contexts, this.filterInput.value?.value);
+    // Let's render the view!
+    return html`
+      <header>
+        <picture class="icon">
+          <!-- User prefers light mode: -->
+          <source
+            srcset="../../assets/logos/logo-dark.svg"
+            media="(prefers-color-scheme: light)"
+          />
+          <!-- User prefers dark mode: -->
+          <source
+            srcset="../../assets/logos/logo-light.svg"
+            media="(prefers-color-scheme: dark)"
+          />
+          <!-- User has no color preference: -->
+          <img src="../../assets/logos/logo-dark.svg" />
+        </picture>
+        <h1>Mozilla VPN</h1>
+      </header>
+      <main>
+        <h2>Website Preferences</h2>
+        <p>
+          You can manage VPN preferences for each website in the Mozilla VPN for
+          Firefox extension.
+        </p>
+        <div>
+          <img />
+          <input
+            type="text"
+            placeholder="Search Websites"
+            ${ref(this.filterInput)}
+            @change=${() => this.requestUpdate()}
+            @input=${() => this.requestUpdate()}
+          />
+        </div>
+        <mz-context-table
+          .contexts=${filteredList}
+          .serverList=${this.serverList}
+          .onRemoveOrigin=${this.removeOrigin.bind(this)}
+        ></mz-context-table>
+      </main>
+    `;
+  }
+  removeOrigin(origin) {
+    this.proxyHandler.then((handler) => {
+      handler.removeContextForOrigin(origin);
+    });
+  }
+
+  static styles = css`
+    ${fontSizing}
+    ${resetSizing}
+
+  body {
+      width: var(--window-width);
+      background-color: var(--panel-bg-color);
+      padding: 0;
+      margin: 0;
+    }
+
+    :host {
+      background-color: var(--panel-bg-color);
+      display: block;
+      width: 100%;
+      height: 100vh;
+      margin: 0;
+      padding: 0;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+    }
+    * {
+      color: var(--text-color-primary);
+    }
+    header {
+      display: flex;
+      flex-direction: row;
+      padding: 20px 30px;
+      border-bottom: 1px solid var(--border-color);
+      width: 100%;
+      margin-bottom: 16px;
+    }
+    main {
+      width: calc(min(70%, 920px));
+    }
+
+    picture img {
+      margin: 0px 5px;
+      height: 30px;
+      width: 30px;
+    }
+
+    h2 {
+      font-size: 38px;
+      font-style: normal;
+      font-weight: 700;
+      line-height: 40px; /* 105.263% */
+      margin-bottom: 18px;
+    }
+    p {
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 400;
+      line-height: 169.336%;
+      margin-bottom: 32px;
+    }
+    input {
+      margin-bottom: 32px;
+      padding: 10px 20px;
+      padding-left: 30px;
+      color: var(--text-color-invert);
```

Comment:
Since it looks like the input background color doesn't change, the text color should stay the same also. In light mode, the input text for me is white and therefore invisible.

Reply:
oh truee, i always forget to test light mode too haha